### PR TITLE
HW34

### PR DESCRIPTION
**Pull request by pavlokhomiak**

### CODE REVIEW

**Comment by ilya449** [comment]
Well done!

**Review by ilya449 — APPROVED** [review]

File: src/main/java/mate/academy/dao/jdbc/UserDaoJdbcImpl.java [hunk 1]
```diff
@@ -46,14 +47,15 @@ public User create(User user) {
...
+        String updateUser = "UPDATE users SET name = ?, login = ?, password = ? salt = ?"
                 + "WHERE user_id = ? AND deleted = false;";
         try (Connection connection = ConnectionUtil.getConnection();
                 PreparedStatement updateUserStatement
                         = connection.prepareStatement(updateUser)) {
             updateUserStatement.setString(1, user.getName());
             updateUserStatement.setString(2, user.getLogin());
             updateUserStatement.setString(3, user.getPassword());
+            updateUserStatement.setBytes(4, user.getSalt());
             updateUserStatement.setLong(4, user.getId());
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/dao/jdbc/UserDaoJdbcImpl.java]:
```suggestion
            updateUserStatement.setBytes(4, user.getSalt());
            updateUserStatement.setLong(5, user.getId());
```

File: src/main/java/mate/academy/dao/jdbc/UserDaoJdbcImpl.java [hunk 2]
```diff
@@ -78,8 +80,10 @@ public User update(User user) {
                 String name = resultSet.getString("name");
                 String login = resultSet.getString("login");
                 String password = resultSet.getString("password");
+                byte[] salt = resultSet.getBytes("salt");
                 User user = new User(name, login, password);
                 user.setId(id);
+                user.setSalt(salt);
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/dao/jdbc/UserDaoJdbcImpl.java]: let's move the repeated logic into the private method `getUserFromResultSet(ResultSet resultSet)`

File: src/main/java/mate/academy/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -17,7 +18,10 @@ public User login(String login, String password)
             throws AuthenticationException {
         Optional<User> userFromDB = userService.findByLogin(login);
 
-        if (userFromDB.isPresent() && userFromDB.get().getPassword().equals(password)) {
+        byte[] salt = userFromDB.get().getSalt();
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/security/AuthenticationServiceImpl.java]: we are calling `get()` on Optional here
but we haven't checked yet if the value is present

File: src/main/resources/init_db.sql [hunk 4]
```diff
@@ -134,5 +134,5 @@ CREATE TABLE `internet_shop`.`users_roles` (
 ENGINE = InnoDB
 DEFAULT CHARACTER SET = utf8;
 
-INSERT INTO users_roles (user_id, role_id) VALUES (1, 2);
-
+ALTER TABLE `internet_shop`.`users`
+ADD COLUMN `salt` VARBINARY(256) NOT NULL AFTER `password`;
```
Comment (KseniiaMakarova) [inline src/main/resources/init_db.sql]: or we can modify CREATE statement

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
everything is good
fix travis please

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**